core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -75,45 +80,130 @@ class BpmBillingProjectLifecycle(
                                  config: MultiCloudWorkspaceConfig,
                                  ctx: RawlsRequestContext
   ): Future[CreationStatus] = {
-    val projectName = createProjectRequest.projectName.value
-    try {
-      val profileModel = blocking {
-        billingProfileManagerDAO.createBillingProfile(projectName, createProjectRequest.billingInfo, ctx)
-      }
+    val projectName = createProjectRequest.projectName
+
+    def createBillingProfile: Future[ProfileModel] =
+      Future(blocking {
+        val profileModel = billingProfileManagerDAO.createBillingProfile(
+          projectName.value,
+          createProjectRequest.billingInfo,
+          ctx
+        )
+        logger.info(
+          s"Creating BPM-backed billing project ${projectName.value}, created profile with ID ${profileModel.getId}."
+        )
+        profileModel
+      })
 
-      val landingZoneResponse = blocking {
-        // This starts a landing zone creation job. There is a separate monitor that polls to see when it
-        // completes and then updates the billing project status accordingly.
+    // This starts a landing zone creation job. There is a separate monitor that polls to see when it
+    // completes and then updates the billing project status accordingly.
+    def createLandingZone(profileModel: ProfileModel): Future[CreateLandingZoneResult] =
+      Future(blocking {
         workspaceManagerDAO.createLandingZone(
           config.azureConfig.get.landingZoneDefinition,
           config.azureConfig.get.landingZoneVersion,
           profileModel.getId,
           ctx
         )
-      }
-      Option(landingZoneResponse.getErrorReport) match {
-        case Some(errorReport) =>
-          throw new LandingZoneCreationException(
-            RawlsErrorReport(StatusCode.int2StatusCode(errorReport.getStatusCode), errorReport.getMessage)
-          )
-        case None =>
-          logger.info(
-            s"Creating BPM-backed billing project ${projectName}, initiated creation of landing zone ${landingZoneResponse.getLandingZoneId} with jobId ${landingZoneResponse.getJobReport.getId}"
-          )
-      }
-      for {
-        _ <- billingRepository.updateLandingZoneId(createProjectRequest.projectName,
-                                                   landingZoneResponse.getLandingZoneId
+      })
+
+    createBillingProfile.flatMap { profileModel =>
+      createLandingZone(profileModel)
+        .flatMap { landingZone =>
+          (for {
+            _ <- Option(landingZone.getErrorReport).traverse { errorReport =>
```

Comment:
@ehigham the `traverse` documentation seems to indicate that it expects something iterable… but obviously it also works here, which is a bit confusing.

Reply:
It's using https://typelevel.org/cats/typeclasses/traverse.html

Reply:
Right, but the documentation doesn't mean much of anything to me… in particular, it seems to imply that the thing you are using traverse with needs to be iterable, and that is not the case here.

Reply:
That type class, `Traverse`, doesn't require the thing to be interable... Scala type classes are a whole other thing and reading them is painful until you've done it enough haha. We can got though that stuff some other time!